WalletWasabi.Fluent/ViewModels/SearchBar/SearchItemProvider.cs

```diff
@@ -39,7 +42,23 @@ private static Func<Task> CreateFunc(NavigationMetaData navigationMetaData)
 				return;
 			}
 
-			Navigate(vm.DefaultTarget).To(vm);
+			if (vm is DialogViewModelBase<Unit> dialog)
+			{
+				if (dialog is AddWalletPageViewModel)
+				{
+					MainViewModel.Instance.IsOobeBackgroundVisible = true;
+					await dialog.NavigateDialogAsync(dialog, NavigationTarget.DialogScreen);
+					MainViewModel.Instance.IsOobeBackgroundVisible = false;
+				}
+				else
+				{
+					await dialog.NavigateDialogAsync(dialog, NavigationTarget.DialogScreen);
+				}
+			}
+			else
+			{
+				Navigate(vm.DefaultTarget).To(vm);
+			}
```

Comment:
And the `SearchItemProvider.Navigate()` can be removed.

```suggestion
			if (vm is NavBarItemViewModel item && item.OpenCommand.CanExecute(default))
			{
				item.OpenCommand.Execute(default);
			}
			else
			{
				RoutableViewModel.Navigate(vm.DefaultTarget).To(vm);
			}
```